### DISCUSSION TITLE
Issues related to MPs deletion

### DISCUSSION THREAD

**Discussion by LiaoMk** [Anura3D/Anura3D_OpenSource#174] [Q&A]
Dear all，

I am currently using the two-point material point method to simulate the seepage process of soil layers, defining high and low water pressure boundaries. During operation, the material points will move outside the calculation grid area. I now want to define a boundary condition so that the material points leaving this boundary are deleted. What do I need to do?

 thanks

**Comment by FrancescaCeccato** [comment]
For inflow/outflow BC I suggest you to read this paper: https://www.sciencedirect.com/science/article/pii/S004579301830728X?casa_token=utTlVIWUXIYAAAAA:tFbLXMjyu8XMSdL7bkfxE2lkW6ys4e5oE5wgc_DuTznAkrEzlp9VwztsErg9ezh4fEPaYRECHlQ
Furthermore, if you only want to remove material points you can adjust the "excavation" procedure.

**Reply by EmreDumlu-UM** [reply]
Hi Francesca, 
Do you think module "AdjustParticleDiscretisation.FOR" is also needed for only removing particles since "MPMExcavation.FOR" uses this module? Because, the module has also a subroutine "AdjustParticleDiscretisation()" where NRemovedParticles is also variable to  remove particles from the discretisation. I just confused when NRemovedParticles exactly defined first.

Thank you for your attention.

**Reply by FrancescaCeccato** [reply]
HI!
NRemovedParticles is a local variable, it is used both in RemoveParticlesFromExcavationElement and in AdjustParticleDiscretisation. it is not a global vatiable.
Note that the subroutine RemoveParticlesFromExcavationElement(IElement) removes all the particles inside the element IElement and then adjust the particle housekeeping. it is called within the subroutine ApplyExcavation() which is applied onlt at the beginning of the load step. In your simulation you probably want it to be called at the beginning/end of each time step...

In your simulation, if you have only liquid MP moving into the elements where you want to remove them (like an outflow boundary condition) you can set these elements as "excavation elements".

**Reply by EmreDumlu-UM** [reply]
Francesca, Thank you very much for your suggestion. It is well suited to our objective in the single-point method.

However, I received same errors several times maybe due to the application of the excavation procedure while working with the double-point method. These errors were seemed once as numerical stability issues, thus I decreased the CFL condition from 0.98 to 0.6. However, they still continued.

The error type was the same always and the indicated element appeared so close to the excavated surface all the time. (Please see the figures below). Red points are solid whereas the greens are liquid particles with subsequent load steps (tL=253->tL=254).  Each load step were specified as 1e-2.  Some solid particles joined the removal process like (18-20), but then the simulation crashed.

![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/6170a2d7-03a3-452a-90e4-f86b3f4baf0d)

![tL_253](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/366b86b1-b923-495c-817b-7ad40f627049)

![tL_254](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/89df8731-6a41-4dd4-8a9a-ecc9ccc66f7a)

**Comment by FrancescaCeccato** [comment]
I am not conviced that the error is due to the excavation... Did you try the same simulation without excavation? If it persists it means that there is another problem. If it disappear it means that this strategy to simulate the outflow is not correct for your problem. One reson may be that with this method you are neglecting the pressure that the outflowing flow applies on the remaining fluid.
